tests/components/geo_location/test_demo.py

```diff
@@ -0,0 +1,87 @@
+"""The tests for the demo platform."""
+import unittest
+
+from homeassistant.components import geo_location
+from homeassistant.components.geo_location.demo import DemoManager, \
+    NUMBER_OF_DEMO_DEVICES, setup_platform
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE
+from homeassistant.core import callback
+from homeassistant.setup import setup_component
+from tests.common import get_test_home_assistant, assert_setup_component
+
+CONFIG = {
+    geo_location.DOMAIN: [
+        {
+            'platform': 'demo'
+        }
+    ]
+}
+
+
+class TestDemoPlatform(unittest.TestCase):
+    """Test the demo platform."""
+
+    def setUp(self):
+        """Initialize values for this testcase class."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    def test_setup_via_config(self):
+        """Test setup of demo platform via configuration."""
+        with assert_setup_component(1, geo_location.DOMAIN):
+            self.assertTrue(setup_component(self.hass, geo_location.DOMAIN,
+                                            CONFIG))
+
+    def test_setup_platform(self):
+        """Test setup of demo platform."""
+        devices = []
+
+        @callback
+        def add_devices_callback(events):
+            """Add recorded devices."""
+            devices.extend(events)
+
+        self.assertTrue(setup_platform(self.hass, None, add_devices_callback,
+                                       None))
+        assert len(devices) == NUMBER_OF_DEMO_DEVICES
+
+    def setup_manager(self):
+        """Setup demo manager."""
+        devices = []
+
+        @callback
+        def add_devices_callback(events):
+            """Add recorded devices."""
+            devices.extend(events)
+
+        return DemoManager(self.hass, add_devices_callback)
+
+    def test_demo_manager(self):
+        """Test demo manager setup."""
+        manager = self.setup_manager()
+        devices = manager._managed_devices
+        assert len(devices) == NUMBER_OF_DEMO_DEVICES
+        # Check a single device's attributes.
+        device = devices[0]
+        self.assertAlmostEqual(device.distance, device.state, places=0)
```

Comment:
Move these assertions to the test above and check the state and state attributes instead.

Reply:
OK